app/src/main/java/com/gh4a/widget/LinkSpan.java

```diff
@@ -39,6 +39,11 @@ private void launchActivity(LinkParser.ParseResult result, FragmentActivity acti
 
     private void openWebPage(Uri clickedUri, FragmentActivity activity) {
         String hostname = clickedUri.getHost();
+        if (hostname == null) {
+            // The user clicked on a relative or partial URL, there's nothing we can do
```

Comment:
Relative URLs should be rewritten to absolute ones in the readme loading logic. Is there any practical example of this happening?

Reply:
Oh, I remember to have seen that logic somewhere in the codebase, but I couldn't find it.
The specific case is https://github.com/TeamNewPipe/NewPipe/, in which anchor links "Screenshots", "Description" under the badges aren't being rewritten (this happens on 4.6.7 too, but doesn't cause any crashes since `URLSpan` handles `ActivityNotFoundException`s).
But really, should anchor links be rewritten? They should ideally scroll down to the anchor, not cause the same file to be reopened.

In any case I would still keep the guard to avoid crashes if URL rewriting or something else doesn't work as expected (like in this case).

Reply:
> Oh, I remember to have seen that logic somewhere in the codebase, but I couldn't find it.
It's [here](https://github.com/slapperwan/gh4a/blob/master/app/src/main/java/com/gh4a/fragment/RepositoryFragment.java#L417)

> But really, should anchor links be rewritten to absolute ones? They should ideally scroll down to the anchor, not cause the same file to be reopened.

I don't see how that scrolling should/could work in our (non-webview) case, so as long as they're present IMHO it's a good idea to let them open the page in the custom tab, scrolled to the right place.

> In any case I would still keep the guard to avoid crashes if URL rewriting or something else doesn't work as expected (like in this case).

Yeah, sure, we can agree on that there should be no crashes ;-)

Reply:
> I don't see how that scrolling should/could work in our (non-webview) case, so as long as they're present IMHO it's a good idea to let them open the page in the custom tab, scrolled to the right place.

The `rewriteRelativeUrls` function you linked above [doesn't explicitly rewrite anchor links](https://github.com/slapperwan/gh4a/blob/ee5b82386b073b5ccff249d02deba94772bfb193/app/src/main/java/com/gh4a/utils/HtmlUtils.java#L157), since it wouldn't work with the current logic: the anchor should be appended to the README path, not to the repository URL as it does now.

Anyway, since this would be an extra improvement, I think that the guard is enough for now.